### PR TITLE
Continue task pipeline if currentState is null

### LINKED ISSUE #1322: Null currentState cause the TASK pipeline to be stuck
### Describe the bug
It has been observed that once participant reconnects, for task related currentState, it carries over currentState as null and requested state to DROPPED. This cause the issue in calculating quota and task pipeline exits with exception.

### To Reproduce
Using base data accessor to set the currentState of the tasks to null.

### Expected behavior
If this happens, the controller should calculate the quota and continue the pipeline.

### PR DESCRIPTION
**Pull request by alirezazamani**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:
Fixes #1322 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
In this PR, if controller encounters null currentStates, it will continue the pipeline and TaskSchedulingStage. Also such currentState would not count toward the quota.

### Tests

- [x] The following tests are written for this issue:
TestTaskCurrentStateNull

- [x] The following is the result of the "mvn test" command on the appropriate module:

Helix-Core:
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestEnableCompression.testEnableCompressionResource:117 expected:<true> but was:<false>
[ERROR]   TestEnqueueJobs.testQueueParallelJobs » ThreadTimeout Method org.testng.intern...
[INFO] 
[ERROR] Tests run: 1175, Failures: 2, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:22 h
[INFO] Finished at: 2020-08-28T11:46:15-07:00
[INFO] ------------------------------------------------------------------------
```

The failed test passed when run individually.
mvn test -Dtest="TestEnableCompression,TestEnqueueJobs"
```
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 79.623 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:25 min
[INFO] Finished at: 2020-08-28T11:48:05-07:00
[INFO] ------------------------------------------------------------------------
```

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/test/java/org/apache/helix/integration/task/TestTaskCurrentStateNull.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    String jobCurrentStatePath2 = "/" + CLUSTER_NAME + "/INSTANCES/" + instanceP0
+        + "/CURRENTSTATES/" + sessionIdP0 + "/" + namespacedJobName2;
+
+    // Read the current states of Participant0 and make sure they have been created
+    boolean isCurrentStateCreated = TestHelper.verify(() -> {
+      ZNRecord recordJob1 = _manager.getHelixDataAccessor().getBaseDataAccessor()
+          .get(jobCurrentStatePath1, new Stat(), AccessOption.PERSISTENT);
+      ZNRecord recordJob2 = _manager.getHelixDataAccessor().getBaseDataAccessor()
+          .get(jobCurrentStatePath2, new Stat(), AccessOption.PERSISTENT);
+      ZNRecord recordTask = _manager.getHelixDataAccessor().getBaseDataAccessor()
```
Comment (NealSun96) [inline helix-core/src/test/java/org/apache/helix/integration/task/TestTaskCurrentStateNull.java:102]: This is the same as recordJob2?

File: helix-core/src/test/java/org/apache/helix/integration/task/TestTaskCurrentStateNull.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * This test makes sure that controller will not be blocked if there exists null current states.
+ */
+public class TestTaskCurrentStateNull extends TaskTestBase {
```
Comment (NealSun96) [inline helix-core/src/test/java/org/apache/helix/integration/task/TestTaskCurrentStateNull.java:44]: I understand that this is meant to be like an Integration test, but would it make more sense to stay in TestAssignableInstanceManager? Or should there be a unit test in TestAssignableInstanceManager?

Comment (alirezazamani) [reply helix-core/src/test/java/org/apache/helix/integration/task/TestTaskCurrentStateNull.java:44]: I prefer to keep as integration test and make sure pipeline runs no matter what currentState is.

File: helix-core/src/test/java/org/apache/helix/integration/task/TestTaskCurrentStateNull.java (hunk 1 above)
Comment (alirezazamani) [reply helix-core/src/test/java/org/apache/helix/integration/task/TestTaskCurrentStateNull.java]: Correct. Fixed. Thanks.

**Comment by alirezazamani** [comment]
This PR is ready to be merged.

Final commit message:
Continue pipeline if currentState is null

In this commit, if controller encounters null currentStates, it will
continue the pipeline and TaskSchedulingStage. Also such 
currentState would not count toward the quota.

**Pull request merged**